### PR TITLE
Rename TFC to HCP Terraform

### PR DESCRIPTION
**Pull request by jpogran**
This renames all user facing information containing the words "Terraform Cloud" to "HCP Terraform".

It does not rename/refactor any code using the old references. The authentication provider ID and sesison key remain the same to preserve existing logins.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,13 +115,13 @@ The extension provides several snippets to accelerate adding Terraform code to y
 - `vare` - Empty variable
 - `varm` - Map Variable
 
-### Terraform Cloud Integration
```
Comment (jpogran) [inline README.md:118]: This will change the markdown links, is that ok?

File: src/providers/authenticationProvider.ts [hunk 2]
```diff
@@ -91,7 +91,9 @@ class TerraformCloudSessionHandler {
   }
 }
 export class TerraformCloudAuthenticationProvider implements vscode.AuthenticationProvider, vscode.Disposable {
-  static providerLabel = 'HashiCorp Terraform Cloud';
+  static providerLabel = 'HashiCorp Cloud Platform Terraform';
+  // These are IDs and session keys that are used to identify the provider and the session in VS Code secret storage
+  // we cannot change these in the rebrand without the user losing the previous session
   static providerID = 'HashiCorpTerraformCloud';
   private sessionKey = 'HashiCorpTerraformCloudSession';
```
Comment (jpogran) [inline src/providers/authenticationProvider.ts:98]: As far as I can tell, changing these will result in the user having to login again.

File: src/terraformCloud/configurationVersion.ts [hunk 3]
```diff
@@ -12,6 +12,8 @@ export const CONFIGURATION_SOURCE: { [id: string]: string } = {
   gitlab: 'GitLab',
   github: 'GitHub',
   terraform: 'Terraform',
+  // TODO: check that the map is rebranded
+  // 'terraform+cloud': 'HCP Terraform',
```
Comment (jpogran) [inline src/terraformCloud/configurationVersion.ts:16]: TODO: Check when/if HCP updates this key

File: src/providers/authenticationProvider.ts (hunk 2 above)
Comment (ansgarm) [reply src/providers/authenticationProvider.ts:98]: Seems fine to me, as this is not user facing :)

File: src/terraformCloud/configurationVersion.ts (hunk 3 above)
Comment (jpogran) [reply src/terraformCloud/configurationVersion.ts:16]: Confirmed this will not change with the rename.

**Review by dbanck — APPROVED** [review]
:shipit:

**Pull request merged**